### ISSUE TITLE
Plugin breaks undopoints

### ISSUE THREAD

**Issue by oblitum** [Chiel92/vim-autoformat#131]
Hi, nice work. Just trying out the plugin and verified the presence this bug that comes up generally with most auto formatters. The following are some pointers about how to fix it, vim-go was first to implement this correctly, which has been adopted on rust.vim:

- https://github.com/fatih/vim-go/issues/40#issuecomment-52426603
- https://github.com/rhysd/vim-clang-format/issues/8#issuecomment-58316885
- https://github.com/rust-lang/rust.vim/pull/52

**Comment by Chiel92** [comment]
Thanks for creating this issue. I'm not entirely sure about the exact behaviour you're referring to. Is it the loss of position when undoing a format action, or is it the case described at https://github.com/fatih/vim-go/issues/40 ?

**Comment by oblitum** [comment]
@Chiel92 yes, I've lost position when undoing format. fatih/vim-go#40 issue covers this, and more, since it was no easy task to get right, Vim doesn't help much, but it got solved.

**Comment by Chiel92** [comment]
Having looked at the implementation in go-fmt for a few minutes, I don't exactly understand all the undofile/undojoin sorcery. Why would you discard the format action from the undohistory (if I understand it correctly)? The problem is not it being there, but the way it handles positions and jumps.

**Comment by zhou13** [comment]
This simple map works for me:
`noremap <F3> ix<ESC>x:undojoin \| Autoformat<CR>`.

Inspired by http://vim.wikia.com/wiki/Restore_the_cursor_position_after_undoing_text_change_made_by_a_script

**Comment by Chiel92** [comment]
Thanks for sharing. This could probably be incorporated in the definition of Autoformat. I don't have much time on my hands currently, but I'll look into it when I have some.

**Comment by keith** [comment]
Related examples here https://github.com/sbdchd/neoformat/issues/134
